### PR TITLE
Add bc-ddpg on Reacher-v2

### PR DESCRIPTION
**Pull request by MrSyee**
Test BC-ddpg on Reacher-v2 (no use HER).
Result is shown in [W&B link](https://app.wandb.ai/khsyee/reinforcement_learning_examples/runs/l7gxqjx9?workspace=user-khsyee)

### CODE REVIEW

**Comment by Curt-Park** [comment]
Please check the wandb link.
It looks like a DQN result.

**Comment by MrSyee** [comment]
> Please check the wandb link.
> It looks like a DQN result.

I fix that. Thank you. :)

**Comment by Curt-Park** [comment]
How about adding gradient clipping or else for stability?

**Comment by MrSyee** [comment]
I redisign HER and Reward function to abstract class.
Code test on LunarLanderContinuous-v2 & Reacher-v2
[W&B](https://app.wandb.ai/khsyee/reinforcement_learning_examples/reports?view=khsyee%2Fbc-her%20test)

File: algorithms/bc/ddpg_agent.py [hunk 1]
```diff
@@ -1,40 +1,61 @@
...
 
+    def __init__(
+        self,
+        env: gym.Env,
+        args: argparse.Namespace,
+        hyper_params: dict,
+        models: tuple,
+        optims: tuple,
+        noise: OUNoise,
+        HER: AbstractHER,
```
Comment (MrSyee) [inline algorithms/bc/ddpg_agent.py]: can use kwargs?

**Comment by MrSyee** [comment]
@Curt-Park [W&B](https://app.wandb.ai/khsyee/reinforcement_learning_examples/reports?view=khsyee%2Fbc-her-test2)

**Pull request merged**